src/table/components/withSelections.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { useContextSelector, TableContext } from '../context';
+import { getSelectionStyle } from '../utils/styling-utils';
+import { getCellSelectionState, SelectionActions, TSelectionActions } from '../utils/selections-utils';
+import { hocProps, SelectionState } from '../../types';
+
+export default function withSelections(CellComponent: (props: hocProps) => JSX.Element) {
```

Comment:
(props: hocProps) => JSX.Element are used twice. 
I think it is better to put it into types.
export type CellComponentProps = (props: hocProps) => JSX.Element

Reply:
sure

Reply:
done